### PR TITLE
OCPBUGS#13156: Updates to the 4.10 Dynamic plugin docs

### PR DESCRIPTION
**Pull request by opayne1**
OCPBUGS#13156: Updates to the 4.10 Dynamic plugin docs

Note to reviewer: This is mainly just a reorg of content and pulls things from the existing 4.12 docs.

Version(s):
4.10 only

Issue:
https://issues.redhat.com/browse/OCPBUGS-13156

Link to docs preview:
https://opayne1.github.io/previews/OCPBUGS-13156/web_console/dynamic-plugins.html

release notes link update: https://opayne1.github.io/previews/OCPBUGS-13156/release_notes/ocp-4-10-release-notes.html#ocp-4-10-dynamic-plugin-technology-preview

QE review:
- [x] QE has approved this change.

Additional information:
The updates include:

- Removal of the old template link
- Updated the "About dynamic plugins" section to match the 4.12 docs
- Removed the "Getting started" section with the "Adding a tab to pods page" example because based on our conversation with PX during 4.12 that is our workaround for being able to have a GH link in the docs.
- Included the Helm chart
- Updated the structure to follow 4.12 for deploying plugin on cluster
- Updated the Additional resources section to remove old links and include the Helm chart docs

### CODE REVIEW

File: modules/about-dynamic-plug-ins.adoc [hunk 1]
```diff
@@ -6,29 +6,41 @@
...
 
-* Prefix your CSS class names with your plug-in name to avoid collisions. For example, `my-plugin_\_heading` and `my-plugin_\_icon`.
+* link:https://nodejs.org/en/[`Node.js`] and link:https://yarnpkg.com/[`yarn`] are required to build and run your plugin.
+* Prefix your CSS class names with your plugin name to avoid collisions. For example, `my-plugin_\_heading` and `my-plugin_\_icon`.
 * Maintain a consistent look, feel, and behavior with other console pages.
-* Use link:https://github.com/openshift/enhancements/blob/master/enhancements/console/dynamic-plugins.md#localization[react-i18next] for localization.
-* _Do not_ use console CSS classes in your markup or override console CSS classes. These are not APIs and are subject to change. Using them might break your plug-in. Avoid selectors like element selectors that could affect markup outside of your plug-in’s components.
+* Follow link:https://www.i18next.com/[react-i18next] localization guidelines when creating your plugin. You can use the `useTranslation` hook like the one in the following example:
++
+[source,ymal]
```
Comment (jhadvig) [inline modules/about-dynamic-plug-ins.adoc:30]:
```suggestion
[source,yaml]
```

File: modules/about-dynamic-plug-ins.adoc [hunk 2]
```diff
@@ -6,29 +6,41 @@
...
+----
+conster Header: React.FC = () => {
```
Comment (jhadvig) [inline modules/about-dynamic-plug-ins.adoc:32]:
```suggestion
const Header: React.FC = () => {
```

File: modules/about-dynamic-plug-ins.adoc (hunk 1 above)
Comment (jhadvig) [reply modules/about-dynamic-plug-ins.adoc:30]: Although this snipped does not contain any yaml, just code snippet

File: modules/about-dynamic-plug-ins.adoc [hunk 3]
```diff
@@ -6,29 +6,41 @@
...
+  const { t } = useTranslation('plugin__console-demo-plugin');
+  return <h1>{t('Hello, World!')}</h1>;
+};
+----
+
+* Avoid selectors that could affect markup outside of your plugins components, such as element selectors. These are not APIs and are subject to change. Using them might break your plugin. Avoid selectors like element selectors that could affect markup outside of your plugins components.
```
Comment (jhadvig) [inline modules/about-dynamic-plug-ins.adoc:38]:
```suggestion
* Avoid selectors that could affect markup outside of your plugin's components, such as element selectors. These are not APIs and are subject to change. Using them might break your plugin. Avoid selectors like element selectors that could affect markup outside of your plugins components.
```

Not a native speaker but does not look right to me

File: modules/deployment-plug-in-cluster.adoc [hunk 4]
```diff
@@ -10,29 +10,86 @@ After pushing an image with your changes to a registry, you can deploy the plugi
```
Comment (jhadvig) [inline modules/deployment-plug-in-cluster.adoc:10]: Should we also change the name of the file to `deployment-plugin-cluster.adoc`

File: modules/about-dynamic-plug-ins.adoc (hunk 1 above)
Comment (opayne1) [reply modules/about-dynamic-plug-ins.adoc:30]: I changed to `[source,text]` to make more generic

File: modules/about-dynamic-plug-ins.adoc (hunk 3 above)
Comment (opayne1) [reply modules/about-dynamic-plug-ins.adoc:38]: I agree there should be an apostrophe there. Fixed

File: modules/deployment-plug-in-cluster.adoc (hunk 4 above)
Comment (opayne1) [reply modules/deployment-plug-in-cluster.adoc]: updated and also updated the others as well. Thanks!

**Comment by opayne1** [comment]
Hi @yapei could you help with a QE review of this PR? I think for the most part it is just reorging the 4.10 docs. I also removed the old template link and replaced with the example we did in 4.12. Thank you! 😄

File: modules/dynamic-plugins-get-started.adoc [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import * as React from 'react';
+
+export default function ExampleTab() {
+    return (
+        <p>This is a custom tab added to a resource using a dynamic plugin.</p>
+    );
+}
+----
+
+.Verification
```
Comment (yapei) [inline modules/dynamic-plugins-get-started.adoc:106]: Shall we also mention that user should deploy && enable the plugin before visiting a Pod page to view the added tab?

Comment (opayne1) [reply modules/dynamic-plugins-get-started.adoc:119]: Yes, thank you! I have added the step with a reference to the doc for further reading.

**Comment by yapei** [comment]
looks good to me, thanks!

**Review by ousleyp — COMMENTED** [review]
Hey Olivia! I have some comments and suggestions for you to consider. I know that you are adding content to 4.10 that already exists in other versions, so it might not make sense to make fixes in this PR. However, I think it would be worth opening one or more follow-up PRs (or maybe filing a bug?) to fix these issues in any versions that they impact.

Please feel free to ping me on Slack if you have any questions about my comments. Thanks!!

**Pull request merged**